src/main/kotlin/treeLib/bintrees/AVLTree.kt

```diff
@@ -0,0 +1,174 @@
+package treeLib.bintrees
+
+import treeLib.bintrees.interfaces.BalancedTree
+import treeLib.bintrees.interfaces.BinTree
+import treeLib.nodes.AVLNode
+
+
+class AVLTree<K : Comparable<K>, V> : BalancedTree<K, V, AVLNode<K, V>>() {
+    override var root: AVLNode<K, V>? = null
+    override var amountOfNodes = 0
+
+    override fun remove(key: K): V? {
+        var value: V? = null
+        fun removeRec(node: AVLNode<K, V>?, key: K): AVLNode<K, V>? {
+            if (node == null) return null
+            if (key < node.key) {
+                node.left = removeRec(node.left, key)
+            } else if (key > node.key) {
+                node.right = removeRec(node.right, key)
+            } else {
+                val nodeA = node.left
+                val nodeB = node.right
```

Comment:
Вот этот момент можно рассмотреть для упрощения. Зачем называть что-то nodeA и nodeB, ведь это нужно запомнить а потом каждый раз проверять точно ли мы запомнили. Сделаем чуть проще назовем переменные тем, что будет в них лежать чуть конкретнее. Примерно так:
```suggestion
                val nodeLeft = node.left
                val nodeRight = node.right
```

Reply:
Да, звучит разумно.